proxy-node-gateway/src/main/java/uk/gov/ida/notification/redis/SessionCacheManager.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.ida.notification.redis;
+
+import io.dropwizard.lifecycle.Managed;
+import uk.gov.ida.notification.exceptions.SessionAlreadyExistsException;
+import uk.gov.ida.notification.exceptions.SessionMissingException;
+import uk.gov.ida.notification.redis.storage.InMemoryStorage;
+import uk.gov.ida.notification.redis.storage.RedisStorage;
+import uk.gov.ida.notification.redis.storage.StorageType;
+import uk.gov.ida.notification.session.GatewaySessionData;
+
+import java.net.URI;
+
+public class SessionCacheManager implements Managed {
+
+    private StorageType storage;
+
+    public SessionCacheManager(URI redisServerURI) {
+        storage = redisServerURI.toString().isEmpty() ?
```

Comment:
Small one; think its pretty much impossible to construct a URI that is empty, would a null check suffice?